lib/src/rewrite.rs

```diff
@@ -846,6 +831,35 @@ pub fn move_commits(
     })
 }
 
+/// Computes the parents of all commits in the connected target set, allowing
+/// only commits in the target set as parents. The parents of each commit are
+/// identical to the ones found using a preorder DFS of the node's ancestors,
+/// starting from the node itself, and avoiding traversing an edge if the
+/// parent is in the target set.
+fn compute_connected_target_commits_internal_parents(
```

Comment:
nit: perhaps, this function can be renamed to `compute_internal_parents_within()` or something? I don't think "connected" is important here. If I understand it, this is the function to remap parents not included in `target_commit_ids` to ancestors by using the graph `connected_target_commits`.